pages/design_documentation/ocf_io_engines.md

```diff
@@ -0,0 +1,292 @@
+---
+title: Cache modes implementation - I/O engines.
+last_updated: August 24, 2022
+toc: false
+permalink: ocf_io_engines.html
+---
+
+Open CAS implements several [**cache modes**](/cache_configuration.html), 
+each suitable for a different usage scenario. To achieve a desired mode of 
+operation, Open CAS Framework uses basic building blocks called **IO engines**. 
+These blocks implement specific algorithms, e.g. write-through or write-back.
+
+
+# OCF I/O engines - general description.
+In general, OCF I/O engines are basic building blocks of a logic behind a 
+particular cache mode. In other words, engines are sets of functions implementing 
+the logic behind write and read operations for a given cache mode 
+(and more, see below).
+
+This is an excerpt of a mapping table used in the OCF to call read or write
+handlers of a particular cache mode (*write-through* and *write-back* modes 
+mappings are shown). 
```

Comment:
Those are not cache modes but io interfaces. Cache modes map to them, but they are not the same. Please clean up the naming and maybe add short explanation of cache mode / io interface / io engine concepts at the beginning.